### PR TITLE
fix: added filtering for empty depName

### LINKED ISSUE #13609: [helm-values] ERROR: Datasource getReleases without lookupName
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Describe the bug

`helm-values` manager does not skip dependencies having a `depName` resolved to an empty string.

Example Repo for reproduction:
https://github.com/schuch/renovate-test/tree/6ec69aa7da4b40ce7cc3012027ba5a635849658d

Related to https://github.com/renovatebot/renovate/discussions/13413

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Matched 1 file(s) for manager helm-values: values.yaml
DEBUG: Found helm-values package files
DEBUG: Found 1 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "helm-values": {
        "fileCount": 1,
        "depCount": 1
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 1
    }
  }
}
ERROR: Datasource getReleases without lookupName
{ ... }



### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by hasanwhitesource**
## Changes:

Call fetchDepUpdates is now not being called when there is any empty string in depName.

## Context:

Closes #13609

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/fetch.ts [hunk 1]
```diff
@@ -50,10 +50,12 @@ async function fetchManagerPackagerFileUpdates(
   const { packageFile } = pFile;
   const packageFileConfig = mergeChildConfig(managerConfig, pFile);
   const { manager } = packageFileConfig;
-  const queue = pFile.deps.map(
-    (dep) => (): Promise<PackageDependency> =>
-      fetchDepUpdates(packageFileConfig, dep)
-  );
+  const queue = pFile.deps
+    .filter((dep) => dep.depName?.length)
```
Comment (viceice) [inline lib/workers/repository/process/fetch.ts:54]: I'm not sure if this is the right place to fix the issue 🤔

Comment (hasanwhitesource) [reply lib/workers/repository/process/fetch.ts:54]: Oh where do you think is the right place? 
I've debugged and saw this is the where the value is coming from (a few calls before getPkgReleases where the error is being logged)

Comment (rarkins) [reply lib/workers/repository/process/fetch.ts:54]: Ideally we could check extracted deps centrally and warn plus add a skip reason if they are missing a valid depName.

Comment (hasanwhitesource) [reply lib/workers/repository/process/fetch.ts:54]: @rarkins Do you think we should just check if the depName is empty and add a skip reason in fetchDepUpdates before line 18 where we check if there is a skip reason for the dep?

**Comment by hasanwhitesource** [comment]
This is the log from running on the repo reproduce data form the linked issue after the change
> DEBUG: Matched 1 file(s) for manager helm-values: values.yaml (repository=hasanwhitesource/renovate-test)                                                                       DEBUG: Found helm-values package files (repository=hasanwhitesource/renovate-test)      DEBUG: Found 1 package file(s) (repository=hasanwhitesource/renovate-test)
> INFO: Dependency extraction complete (repository=hasanwhitesource/renovate-test)              "baseBranch": "master",                "stats": {                                                                                "managers": {"helm-values": {"fileCount": 1, "depCount": 1}},                           "total": {"fileCount": 1, "depCount": 1}
>        }
> DEBUG: Package releases lookups complete (repository=hasanwhitesource/renovate-test)
>        "baseBranch": "master"
> DEBUG: branchifyUpgrades (repository=hasanwhitesource/renovate-test)
> DEBUG: 0 flattened updates found:  (repository=hasanwhitesource/renovate-test)
> DEBUG: Returning 0 branch(es) (repository=hasanwhitesource/renovate-test)
> DEBUG: config.repoIsOnboarded=true (repository=hasanwhitesource/renovate-test)
> DEBUG: packageFiles with updates (repository=hasanwhitesource/renovate-test)
>        "config": {
>          "helm-values": [
>            {
>              "packageFile": "values.yaml",
>              "deps": [
>                {
>                  "depName": "",
>                  "currentValue": "2.8.11",
>                  "datasource": "docker",
>                  "replaceString": "2.8.11",
>                  "versioning": "docker",
>                  "autoReplaceStringTemplate": "{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}",
>                  "depIndex": 0,
>                  "updates": [],
>                  "skipReason": "empty"
>                }
>              ]
>            }
>          ]
>        }
> DEBUG: processRepo() (repository=hasanwhitesource/renovate-test)

File: lib/workers/repository/process/fetch.ts [hunk 2]
```diff
@@ -15,6 +15,9 @@ async function fetchDepUpdates(
 ): Promise<PackageDependency> {
   let dep = clone(indep);
   dep.updates = [];
+  if (dep.depName?.length === 0) {
```
Comment (rarkins) [inline lib/workers/repository/process/fetch.ts:18]:
```suggestion
  if (!is.nonEmptyString(dep.depName)) {
```

File: lib/workers/repository/process/fetch.ts [hunk 3]
```diff
@@ -15,6 +15,9 @@ async function fetchDepUpdates(
...
+    dep.skipReason = 'empty';
```
Comment (rarkins) [inline lib/workers/repository/process/fetch.ts:19]: I think we need to introduce a more explanatory skip reason, like `missing-depname`

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/process/fetch.ts [hunk 4]
```diff
@@ -15,6 +16,9 @@ async function fetchDepUpdates(
 ): Promise<PackageDependency> {
   let dep = clone(indep);
   dep.updates = [];
+  if (!is.nonEmptyString(dep.depName)) {
```
Comment (rarkins) [inline lib/workers/repository/process/fetch.ts:19]:
```suggestion
  if (!is.nonEmptyString(dep.depName?.trim)) {
```

Please add tests for `null`, `undefined` and `' '`, `''` to make sure this is behaving as expected :)

**Review by rarkins — CHANGES_REQUESTED** [review]
trim before check

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Now I want to double check something. @JamieMagee @zharinov can you recall any cases where we *intentionally* have an empty `depName`?

**Comment by zharinov** [comment]
Can't remember any such case 🤷‍♂️

**Comment by hasanwhitesource** [comment]
@rarkins any follow ups here ?
